pkg/apis/tf/v1alpha1/terraform_types.go

```diff
@@ -136,21 +136,6 @@ type TerraformSpec struct {
 	// providers
 	Credentials []Credentials `json:"credentials,omitempty"`
 
-	// ApplyOnCreate is used to apply any planned changes when the resource is
-	// first created. Omitting this or setting it to false will resort to
-	// on demand apply. Defaults to false.
-	ApplyOnCreate bool `json:"applyOnCreate,omitempty"`
-
-	// ApplyOnUpdate is used to apply any planned changes when the resource is
-	// updated. Omitting this or setting it to false will resort to
-	// on demand apply. Defaults to false.
-	ApplyOnUpdate bool `json:"applyOnUpdate,omitempty"`
```

Comment:
Why is this property removed? I am planning to implement a similar feature of this operator or Crossplane in https://github.com/oam-dev/terraform-controller. If Terraform configuration is the single source of truth, it would be great for the operator/controller to periodically check the external cloud resource is changed from Dashboard, outside OpenAPI.

Reply:
I didn't like the mechanism that was used to move forward with "terraform apply" when this field was true. It blocked the run until a user updated a configmap with a special keyword. In essence, it was a clunky and poorly documented design. Even I forgot how to continue with apply at times. 

The alternative I'll propose is using the postPlanScript to pause the workflow. The user can determine what pauses it, and how to unpause it. 

I'm currently working heavily on documentation and I will probably post a blog on these kind of use cases. 

On another note, I was reading up on your project, https://github.com/oam-dev/terraform-controller and I like how providers secrets are first class citizens.

Reply:
`postPlanScript` is an brilliant idea. 

To help me better clarify this feature, do the changes come from Terraform Configuration or applied cloud resources. I mean the latter one, but it seems the former per your explanation.

The design of the first-class citizens of Provider is inspired by Crossplane. We cooperated a lot and I also noted in the Design doc:)